### PR TITLE
NvHost: Fix some regressions and correct signaling on timeout.

### PR DESCRIPTION
**Pull request by FernandoS27**
This is a series of fixes half fixing regressions and other half suggested by ByLaws from skyline emulation.

### CODE REVIEW

**Comment by bunnei** [comment]
Build failures are real

File: src/core/hle/service/nvdrv/devices/nvhost_ctrl.cpp [hunk 1]
```diff
@@ -106,30 +106,28 @@ NvResult nvhost_ctrl::IocCtrlEventWait(const std::vector<u8>& input, std::vector
...
     auto& gpu = system.GPU();
-
-    // This is mostly to take into account unimplemented features. As synced
-    // gpu is always synced.
-    if (!gpu.IsAsync()) {
-        event.event->GetWritableEvent().Signal();
-        return NvResult::Success;
-    }
     const u32 current_syncpoint_value = event.fence.value;
     const s32 diff = current_syncpoint_value - params.threshold;
```
Comment (v1993) [inline src/core/hle/service/nvdrv/devices/nvhost_ctrl.cpp:110]:
```
../src/core/hle/service/nvdrv/devices/nvhost_ctrl.cpp:110:15: error: unused variable ‘diff’ [-Werror=unused-variable]
  110 |     const s32 diff = current_syncpoint_value - params.threshold;
      |               ^~~~

```

Comment (liberodark) [reply src/core/hle/service/nvdrv/devices/nvhost_ctrl.cpp:110]: Same for me.

File: src/core/hle/service/nvdrv/nvdrv.h [hunk 2]
```diff
@@ -50,7 +50,7 @@ struct EventInterface {
     // Tells if an NVEvent is registered or not
     std::array<bool, MaxNvEvents> registered{};
     // Tells the NVEvent that it has failed.
```
Comment (v1993) [inline src/core/hle/service/nvdrv/nvdrv.h:52]: Comment needs an update.

**Comment by FernandoS27** [comment]
Should now be fixed.

**Comment by FernandoS27** [comment]
I realised that ctrl is not working correctly. I'll work harder to rewrite more of it.

**Comment by FernandoS27** [comment]
Please don't tag this PR.

**Comment by FernandoS27** [comment]
I have decided to close it and rework everything along other changes. It will be the best.

**Pull request closed without merging**